### PR TITLE
feat(argo): add composition for argo

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
add composition for argo

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested
```
--- PASS: kuttl (139.55s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (139.05s)
PASS
00:38:25 [ OK ] running automated tests
```

### CODE REVIEW

File: apis/composition.yaml [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+              controller:
+                replicas: 1
+                metrics:
+                  enabled: true
+                  service:
+                    annotations:
+                      prometheus.io/scrape: true
+                env:
+                - name: ARGOCD_K8S_CLIENT_QPS
+                  value: "300"
```
[the remark this answers is not in the document]
Comment (haarchri) [reply apis/composition.yaml:52]: in readme or what is your idea ?

File: apis/composition.yaml [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+                  application.namespaces: "cluster-*"
+                cm:
+                  application.resourceTrackingMethod: "annotation"
+                  resource.exclusions: |
+                    - kinds:
+                      - ProviderConfigUsage
+                      apiGroups:
+                      - "*"
+                  resource.customizations: |
+                    "*.upbound.io/*":
```
[the remark this answers is not in the document]
Comment (haarchri) [reply apis/composition.yaml:105]: in readme or what is your idea ?

[the remark this answers is not in the document]
Comment (ytsarev) [reply apis/composition.yaml:105]: inline.

File: apis/composition.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ytsarev) [reply apis/composition.yaml:52]: inline should be sufficient.

**Pull request merged**